manifests/init.pp

```diff
@@ -224,9 +224,12 @@
   }
 
 
-  # Configure default gateway (On RedHat). Also hostname is set.
+  # Configure sysconfig/network (On RedHat).
   if $::osfamily == 'RedHat'
-  and $network::gateway {
+    and ($network::gateway
+    or $network::hostname
+    or $network::ipv6enable
+    or $network::nozeroconf) {
```

Comment:
Why do we need to check for parameters at all?
The template itself checks for parameters and values.
We only do this on RedHat. Is there a specific reason for doing so?